File: mgr-backend-node/algorithms/primeNumbers.js
function checkIfNumbersPrimeInRange(number) {
    let outputStr = '';

    for (let i = 2; i <= number; i++) {
        let isPrime = true;

        for (let j = 2; j < i; j++) {
            if (i % j === 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime === true) {
            outputStr += i + ' ';
        }
    }
    return outputStr;
}

let t0 = process.hrtime();
checkIfNumbersPrimeInRange(300000);
let t1 = process.hrtime(t0);
console.log(`Execution time: ${t1[0]}s ${t1[1] / 1000000}ms`);

module.exports = {checkIfNumbersPrimeInRange};

File: mgr-backend-spring/src/main/java/org/bszabat/mgrbackend/helpers/PrimeNumbers.java
//package org.bszabat.mgrbackend.algorithms;
//
//import org.springframework.stereotype.Component;
//
//@Component
//public class PrimeNumbers {
//
//    public static String checkIfNumbersPrimeInRange(int number) {
//        if (number < 2) {
//            return "No prime numbers in given range";
//        }
//
//        String outputStr = "";
//        for (int i = 2; i <= number; i++) {
//            boolean isPrime = true;
//
//            for (int j = 2; j < i; j++) {
//                if(i % j == 0) {
//                    isPrime = false;
//                    break;
//                }
//            }
//            if (isPrime) {
//                outputStr += i + " ";
//            }
//        }
//        return outputStr.substring(0, outputStr.length()-1);
//    }
//
//    public static void main(String[] args) {
//        long startTime = System.nanoTime();
//        PrimeNumbers.checkIfNumbersPrimeInRange(300000);
//        long endTime = System.nanoTime();
//        long elapsedTimeMs = (endTime - startTime) / 1000000;
//        System.out.println(String.format("Execution time: %ss %sms", elapsedTimeMs/1000, elapsedTimeMs%1000));
//    }
//}
